src/filters/date/money/money.ts

```diff
@@ -0,0 +1,31 @@
+import Vue, { PluginObject } from 'vue';
+import MessagePlugin from '../../../components/message/message';
+import { MMoney } from '../../../utils/money/money';
+import { ModulVue } from '../../../utils/vue/vue';
+import { MONEY_NAME } from '../../filter-names';
+
+export class MoneyFilter {
+    static formatCurrency(money: MMoney): string {
```

Comment:
On ne devrait pas suivre l'API des filtres de Vue (
https://fr.vuejs.org/v2/guide/filters.html) et exporter une fonction?

Reply:
où est-ce que c'est écrit qu'un filter doit prendre en paramètre un string?

Reply:
```
capitalize: function (value) {
    if (!value) return ''
    value = value.toString()
    return value.charAt(0).toUpperCase() + value.slice(1)
  }
```

Ici value est converti en string par la fonction... ce qui présuppose que value n'est pas nécéssairement un string

Reply:
Ouais c'est peut être pas explique que ca doit être une string . Mais le filtre doit être utilisable par la syntaxt suivante dans les template: 

`{{ 233 | f-m-money}}`

Reply:
1- alors pourquoi on accepte de faire un filtre pour les périodes?
2- Dans ce cas ça prendrait plus un filtre de number-as-money.  Money est un type complexe, c'est pas un type built-in.  Pour afficher un monnaie ça prend un nombre et une currency.
3- Sinon j'enlève le filter pis je garde juste la fonction... mais rendu-là c'est juste dommage de ne pas exposer la fonction comme filtre puisque ça va amener d'la duplication de code partout (définir un getter pour la valeur en string, importer la fonction, appeller la fonction avec le money)

Reply:
![image](https://user-images.githubusercontent.com/18729439/54696364-0d7fae00-4b02-11e9-891b-ac834aeb1f3d.png)
Ici un filtre qui prend en paramètre un objet et l'output en json..
Y'en a un autre qui output les keys de built-in dans vue aussi.

Reply:
Ouais je viens de voir le filtre des période 💩 

Pour ce qui est des paramétré avec un filtre la facon de faire est d'exposer 2 argument a la fonction. 

```
{{ 233 | f-m-money('CAD')}} = $ 233.00
{{ 233 | f-m-money('EUR')}} = E 233.00
```

De cette façon le filtre est beaucoup plus réutilisable et conforme a l'api de vue sans qu'on soit obliger de faire une syntaxe avec $options dans les template (ce qui est très obscure pour un néophyte)

Reply:
Je ne suis pas d'accord avec ton argument de réutilisabilité.  Les monnaies ne sont pas représentés par des number puisque c'est une définition insufisante.  Ça nous prend un type custom de monnaie.

Je ne vois pas dans quel cas c'est utile de faire comme tu propose de faire.  

Alors que je suis certain que de devoir faire
{{ money.amount | filter(money.currency) }}
 ça devient chiant rapidement.

Nos filtres devraient se rapprocher de notre implémentation de monnaie.

Reply:
Sinon je peux ramener tout ça dans brio, closer ma review, delete ma branche pcq nos besoins sont clairement différents :)

Reply:
> $options dans les template (ce qui est très obscure pour un néophyte)

@chuckmah C'est que pour utiliser le filtre dans un v-html. Il n'y a pas de lien avec les paramètres entrants.

Reply:
merci @Netnix  je viens de comprend d'ou ca vient les $options (https://github.com/vuejs/vue/issues/4352) 
voir la réponse de Evan You :neckbeard: 

Mais ça reste un hack . Dans modul, on devrait garder la définition original des filtres pour permettre de les utiliser avec des moustaches et de pouvoir les piper.

Reply:
@Mboulianne  j'en ai glissé un mot avec @simardo et effectivement ca fait plus de sens que vous ramener le type Money dans Brio. Dans modul on va traiter le currency en number